include/deepworks/parameter.hpp

```diff
@@ -0,0 +1,27 @@
+#pragma once
+
+#include <deepworks/tensor.hpp>
+
+namespace deepworks {
+
+class Parameter {
+public:
+    Parameter(Tensor&& data, bool is_trainable=true);
+
+    const Tensor& data() const;
+          Tensor& data();
+
+    const Tensor& grad() const;
+          Tensor& grad();
+
+    void train(bool mode);
+    bool is_trainable() const;
```

Comment:
Do we need to make the flag a tensor too?

now we turn off/on the whole layer?

Reply:
No, because for instance: `Linear` layer has two parameterers for `weight` and `bias`, so we can freeze only `weight`